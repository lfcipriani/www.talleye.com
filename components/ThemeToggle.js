import React from 'react';
import { ThemeContext } from './Theme';

export default class ThemeToggle extends React.Component {
  constructor(props) {
    super(props);
  }

  sunIcon() {
    return (
      <>
        <span className="visually-hidden">Toggle to dark theme</span>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="#000000"
          viewBox="0 0 256 256"
          width={this.props.size || '35'}
          height={this.props.size || '35'}
          className="svgIcon"
          aria-hidden="true"
          focusable="false"
        >
          <rect width="256" height="256" fill="none"></rect>
          <circle
            cx="128"
            cy="128"
            r="60"
            fill="none"
            stroke="#000000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></circle>
          <line
            x1="128"
            y1="36"
            x2="128"
            y2="28"
            fill="none"
            stroke="#000000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></line>
          <line
            x1="62.94618"
            y1="62.94618"
            x2="57.28932"
            y2="57.28932"
            fill="none"
            stroke="#000000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></line>
          <line
            x1="36"
            y1="128"
            x2="28"
            y2="128"
            fill="none"
            stroke="#000000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></line>
          <line
            x1="62.94618"
            y1="193.05382"
            x2="57.28932"
            y2="198.71068"
            fill="none"
            stroke="#000000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></line>
          <line
            x1="128"
            y1="220"
            x2="128"
            y2="228"
            fill="none"
            stroke="#000000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></line>
          <line
            x1="193.05382"
            y1="193.05382"
            x2="198.71068"
            y2="198.71068"
            fill="none"
            stroke="#000000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></line>
          <line
            x1="220"
            y1="128"
            x2="228"
            y2="128"
            fill="none"
            stroke="#000000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></line>
          <line
            x1="193.05382"
            y1="62.94618"
            x2="198.71068"
            y2="57.28932"
            fill="none"
            stroke="#000000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></line>
        </svg>
        <style jsx>
          {`
            .stroke {
              stroke: var(--color-icon);
            }
            .fill {
              fill: var(--color-icon);
            }
          `}
        </style>
      </>
    );
  }

  moonIcon() {
    return (
      <>
        <span className="visually-hidden">Toggle to light theme</span>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width={this.props.size || '35'}
          height={this.props.size || '35'}
          fill="#284B63"
          viewBox="0 0 256 256"
          className="svgIcon"
          aria-hidden="true"
          focusable="false"
        >
          <rect width="256" height="256" fill="none"></rect>
          <path
            d="M88,148a68,68,0,1,1,68,68H76a44,44,0,1,1,14.30583-85.62208"
            fill="none"
            stroke="#284B63"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></path>
          <path
            d="M46.99,138.92889A64.12919,64.12919,0,0,1,9.6218,94.37955l.00023-.001A64.0566,64.0566,0,0,0,86.37854,17.622l.00084-.00019A64.02478,64.02478,0,0,1,136,80q0,1.51431-.06961,3.01174"
            fill="none"
            stroke="#284B63"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></path>
        </svg>
        <style jsx>
          {`
            .stroke {
              stroke: var(--color-icon);
            }
            .fill {
              fill: var(--color-icon);
            }
          `}
        </style>
      </>
    );
  }

  render() {
    return (
      <ThemeContext.Consumer>
        {({ theme, toggleTheme }) => (
          <a href="#" onClick={toggleTheme}>
            {theme === 'light' ? this.moonIcon() : this.sunIcon()}
          </a>
        )}
      </ThemeContext.Consumer>
    );
  }
}
