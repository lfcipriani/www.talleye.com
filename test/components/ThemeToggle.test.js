import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ThemeToggle from '../../components/ThemeToggle';
import ThemeProvider from '../../components/Theme';

beforeAll(() => {
  const useRouter = jest.spyOn(require('next/router'), 'useRouter');
  useRouter.mockImplementation(() => ({
    locale: 'en',
    defaultLocale: 'en',
  }));
});

test('renders the element', () => {
  const component = renderer.create(<ThemeToggle />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders the icon in the specified size', () => {
  const component = renderer.create(<ThemeToggle size="42" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('ThemeToggle', () => {
  beforeEach(() => {
    render(
      <ThemeProvider>
        <ThemeToggle />
      </ThemeProvider>
    );
  });

  it('renders light theme by default', () => {
    const icon = screen.getByText(/Toggle to dark theme/);
    expect(icon).toBeInTheDocument();
  });

  it('switches icon when link is clicked', () => {
    var icon = screen.getByText(/Toggle to dark theme/);
    fireEvent.click(icon);

    icon = screen.getByText(/Toggle to light theme/);
    expect(icon).toBeInTheDocument();
  });
});
