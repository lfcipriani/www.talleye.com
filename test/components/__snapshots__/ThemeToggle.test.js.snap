// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the element 1`] = `
<a
  href="#"
  onClick={[Function]}
  title="Toggle to dark theme"
>
  <span
    className="jsx-676870541 visually-hidden"
  >
    Toggle to dark theme
  </span>
  <svg
    aria-hidden="true"
    className="jsx-676870541 svgIcon"
    fill="#284B63"
    focusable="false"
    height="35"
    viewBox="0 0 256 256"
    width="35"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      className="jsx-676870541"
      fill="none"
      height="256"
      width="256"
    />
    <path
      className="jsx-676870541 stroke"
      d="M88,148a68,68,0,1,1,68,68H76a44,44,0,1,1,14.30583-85.62208"
      fill="none"
      stroke="#284B63"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <path
      className="jsx-676870541 stroke"
      d="M46.99,138.92889A64.12919,64.12919,0,0,1,9.6218,94.37955l.00023-.001A64.0566,64.0566,0,0,0,86.37854,17.622l.00084-.00019A64.02478,64.02478,0,0,1,136,80q0,1.51431-.06961,3.01174"
      fill="none"
      stroke="#284B63"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
  </svg>
</a>
`;

exports[`renders the icon in the specified size 1`] = `
<a
  href="#"
  onClick={[Function]}
  title="Toggle to dark theme"
>
  <span
    className="jsx-676870541 visually-hidden"
  >
    Toggle to dark theme
  </span>
  <svg
    aria-hidden="true"
    className="jsx-676870541 svgIcon"
    fill="#284B63"
    focusable="false"
    height="42"
    viewBox="0 0 256 256"
    width="42"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      className="jsx-676870541"
      fill="none"
      height="256"
      width="256"
    />
    <path
      className="jsx-676870541 stroke"
      d="M88,148a68,68,0,1,1,68,68H76a44,44,0,1,1,14.30583-85.62208"
      fill="none"
      stroke="#284B63"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <path
      className="jsx-676870541 stroke"
      d="M46.99,138.92889A64.12919,64.12919,0,0,1,9.6218,94.37955l.00023-.001A64.0566,64.0566,0,0,0,86.37854,17.622l.00084-.00019A64.02478,64.02478,0,0,1,136,80q0,1.51431-.06961,3.01174"
      fill="none"
      stroke="#284B63"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
  </svg>
</a>
`;
